### PR TITLE
Category

### PR DESCRIPTION
**Pull request by DuaaH**

### CODE REVIEW

File: server/database/queries/getCategory.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const dbConnection = require("./../db_connection")
+
+const getCategoryInfo = () => {
+  console.log("query")
```
Comment (Muniralsharif) [inline server/database/queries/getCategory.js]: remove console.log :)

File: .gitignore [hunk 2]
```diff
@@ -56,9 +56,9 @@ typings/
...
 .env
-
+config.env
 # next.js build output
 .next
 .node_modules
 /client/build
-config.env
\ No newline at end of file
+config.env
```
Comment (ishak52) [inline .gitignore]: No need to ignore `config.env` two times in the same file

File: client/src/components/Category/index.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    })
+    return arr2.length
+  }
+
+  render() {
+    const { data } = this.state.details
+    const { userPledges } = this.state.userPledges
+    const { pledgesCategory } = this.state.categories
+    return (
+      <switch>
```
Comment (ishak52) [inline client/src/components/Category/index.js]: Used `Switch` but never defined, you should import it first 
`import {Switch} from "react-router-dom";`
also when you use the tag `<Switch>` use it capitalized

File: client/src/components/NumberOfPledges/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+class NumberOfPledges extends Component {
+  state = {
+    details: [],
+    allPledges: []
+  }
+  componentDidMount = () => {
+    const userId = 1
+    axios.get(`/api/dashboard_number_of_pledges/${userId}`).then(({ data }) => {
+
+
```
Comment (ishak52) [inline client/src/components/NumberOfPledges/index.js]: **Indentation**
Delete extra white lines

File: client/src/components/Category/style.css [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+.category-div__dashboard-action{ 
+    width:100%;
+    height: 145px;
+    font-style: Raleway;
+    position: relative;
+    margin-top:18px;   
+    margin-left: 5%;
+    margin-right: 5%
+}
+.category-div__dashboard-action__image{
```
Comment (ishak52) [inline client/src/components/Category/style.css]: Use BEM styling in the write way

File: server/database/queries/getNumberPledgesCategory.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const dbConnection = require('./../db_connection')
+
+const getNumberPledgesCategory = (categoryName) => {
+  return dbConnection
+    .query('select * from pledge inner join category on pledge.category_id =category.category_id ')
```
Comment (ishak52) [inline server/database/queries/getNumberPledgesCategory.js]: Capitalize the query statements like `SELECT, FROM, ...`

File: client/src/components/Category/index.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+    history.push({
+      pathname: `action-category/${id}`,
+      data: this.state.details
+    })
+  }
+
+  // function to calculate all pledges that user make it in category by filter
+  numberOfPledges = (arr1, item) => {
+    let arr2 = arr1.filter((element) => {
```
Comment (ishak52) [inline client/src/components/Category/index.js]: Expected to return a value at the end of the filter function

File: client/src/components/Category/index.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      if (element.name === item) {
+        return element.pledge_name
+      }
+    })
+    return arr2.length
+  }
+
+  //function to calculate all pledges in the category by filtering the array
+  numberOfPledges2 = (arr1, item) => {
+    let arr2 = arr1.filter((element) => {
```
Comment (ishak52) [inline client/src/components/Category/index.js]: Expected to return a value at the end of the filter function

File: .gitignore (hunk 2 above)
Comment (DuaaH) [reply .gitignore]: delete one statement from .gitignore

File: client/src/components/Category/index.js (hunk 3 above)
Comment (DuaaH) [reply client/src/components/Category/index.js]: done

File: client/src/components/NumberOfPledges/index.js (hunk 4 above)
Comment (DuaaH) [reply client/src/components/NumberOfPledges/index.js]: done

File: client/src/components/Category/style.css (hunk 5 above)
Comment (DuaaH) [reply client/src/components/Category/style.css]: done

File: server/database/queries/getNumberPledgesCategory.js (hunk 6 above)
Comment (DuaaH) [reply server/database/queries/getNumberPledgesCategory.js]: done

File: client/src/components/Category/index.js (hunk 7 above)
Comment (DuaaH) [reply client/src/components/Category/index.js]: done

File: client/src/components/Category/index.js (hunk 8 above)
Comment (DuaaH) [reply client/src/components/Category/index.js]: done

**Pull request merged**